blocks/reader-revenue-manager/contribute-with-google/Edit.js

```diff
@@ -0,0 +1,138 @@
+import { CORE_MODULES } from '../../../assets/js/googlesitekit/modules/datastore/constants';
+import { MODULES_READER_REVENUE_MANAGER } from '../../../assets/js/modules/reader-revenue-manager/datastore/constants';
+
+/**
+ * Contribute with Google Block Edit component.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} props        Component props.
+ * @param {Object} props.select Data store select function.
+ * @return {Element} Element to render.
+ */
+export default function Edit( { select } ) {
+	const blockProps = useBlockProps();
+
+	const hasModuleAccess = select( CORE_MODULES ).hasModuleOwnershipOrAccess(
+		'reader-revenue-manager'
+	);
+
+	const settings = select( MODULES_READER_REVENUE_MANAGER ).getSettings();
+
+	const { publicationID, paymentOption, snippetMode, postTypes } = settings;
+
+	const serviceURL = select( MODULES_READER_REVENUE_MANAGER ).getServiceURL( {
+		path: 'reader-revenue-manager',
+		query: {
+			publication: publicationID,
+		},
+	} );
+
+	const metaKey = `googlesitekit_rrm_${ publicationID }:productID`;
+
+	const postProductID =
+		select( CORE_EDITOR ).getEditedPostAttribute( 'meta' )?.[ metaKey ] ||
+		'';
+	const postType = select( CORE_EDITOR ).getCurrentPostType();
+
+	let notice = '';
+	let disabled = false;
+
+	if ( paymentOption !== 'contributions' ) {
+		disabled = true;
+
+		if ( hasModuleAccess ) {
+			notice = createInterpolateElement(
+				__(
+					'You need to set up a contributions request in Reader Revenue Manager to use this block. <a>Go to Reader Revenue Manager</a>',
+					'google-site-kit'
+				),
+				{
+					a: <ExternalLink href={ serviceURL } />,
+				}
+			);
+		} else {
+			notice = __(
+				'You need to set up a contributions request in Reader Revenue Manager to use this block. Contact your administrator.',
+				'google-site-kit'
+			);
+		}
+	}
+
+	if (
+		postProductID === 'none' ||
+		( ! postProductID && snippetMode === 'per_post' ) ||
+		( ! postProductID &&
+			snippetMode === 'post_types' &&
+			! postTypes.includes( postType ) )
+	) {
+		disabled = true;
+
+		if ( hasModuleAccess ) {
+			notice = createInterpolateElement(
+				__(
+					'This post does not include the Reader Revenue Manager snippet. Configure the snippet for this post in the post settings sidebar.',
+					'google-site-kit'
+				),
+				{
+					a: <ExternalLink href={ serviceURL } />,
+				}
+			);
+		} else {
+			notice = __(
+				'This post does not include the Reader Revenue Manager snippet. Contact your administrator',
+				'google-site-kit'
+			);
+		}
+	}
+
+	return (
+		<Fragment>
+			{ notice && (
+				<InspectorControls>
+					<div className="block-editor-block-card">
+						<Notice status="warning" isDismissible={ false }>
+							{ notice }
+						</Notice>
+					</div>
+				</InspectorControls>
+			) }
+			<div { ...blockProps }>
+				<div className="googlesitekit-blocks-reader-revenue-manager">
+					<EditorButton disabled={ disabled }>
+						Contribute with Google
```

Comment:
@nfmohit, just to be clear is the expectation that the real button content will always be this text in English?

Reply:
Good question. In the actual button that renders in the front end, the text is populated by the `swg-basic.js` library, so it is not translatable, not at least from our end. The `swg-basic.js` library does populate the button with text using a different language if the site language is different, but since we don't load `swg-basic.js` in the editor, we can't replicate that. Do you think we should make it translatable for our editor preview?

Reply:
Thanks @nfmohit. Seeing as the actual button will be rendered with text in the site's language, I think we should do our best to translate it in the editor too.

I'd suggest making the text translatable and linking to the list of titles for the relevant button in the `translators` hint, e.g. [here](https://github.com/subscriptions-project/swg-js/blob/05af2d45cfcaf831a6b4d35c28f2c7b5c2e39308/src/i18n/swg-strings.ts#L58-L91) for the Contribute button.

The styling will also need modifying to better handle variable width text.

Reply:
I agree, I've updated the button accordingly, thanks Tom!